src/languageTools/DefaultProviders.js

```diff
@@ -109,49 +109,51 @@ define(function (require, exports, module) {
             $deferredHints = $.Deferred(),
             self = this;
 
-        this.client.requestHints({
-            filePath: docPath,
-            cursorPos: pos
-        }).done(function (msgObj) {
-            var context = TokenUtils.getInitialContext(editor._codeMirror, pos),
-                hints = [];
-
-            self.query = context.token.string.slice(0, context.pos.ch - context.token.start);
-            if (msgObj) {
-                var res = msgObj.items,
-                    filteredHints = filterWithQueryAndMatcher(res, self.query);
-
-                StringMatch.basicMatchSort(filteredHints);
-                filteredHints.forEach(function (element) {
-                    var $fHint = $("<span>")
-                        .addClass("brackets-hints");
-
-                    if (element.stringRanges) {
-                        element.stringRanges.forEach(function (item) {
-                            if (item.matched) {
-                                $fHint.append($("<span>")
-                                    .append(_.escape(item.text))
-                                    .addClass("matched-hint"));
-                            } else {
-                                $fHint.append(_.escape(item.text));
-                            }
-                        });
-                    } else {
-                        $fHint.text(element.label);
-                    }
+        setTimeout(function(){
```

Comment:
@niteskum you should just send the document change event explicitly from here, which will ensure that the aync calls enter the event queue in the correct order. setTimeout doesn't guarantee that requestHints will always happen after notifyTextDocumentChanged. 

Just paste the snippet below above the requestHints call.
``` 
//Make sure the document is in sync with the server
this.defaultCodeHintProviders.client.notifyTextDocumentChanged({
    filePath: docPath,
    fileContent: editor.document.getText()
});
```

Reply:
changed , please check this once in your machine also.